signalfx/resource_signalfx_aws_integration.go

```diff
@@ -678,78 +757,38 @@ func DoIntegrationAWSDelete(d *schema.ResourceData, meta interface{}) error {
 	// Disable the AWS logs sync and/or CloudWatch metric streams sync if needed
 	needToDisableMetricStreams := int.Enabled && int.MetricStreamsSyncState != "" && int.MetricStreamsSyncState != "DISABLED"
 	needToDisableLogsSync := int.Enabled && int.LogsSyncState != "" && int.LogsSyncState != "DISABLED"
+	var toWaitFor []TargetState
 	if needToDisableMetricStreams || needToDisableLogsSync {
 		if needToDisableMetricStreams {
 			int.MetricStreamsSyncState = "CANCELLING"
+			toWaitFor = append(toWaitFor, TargetState{"metric_streams_sync_state", "disabled"})
 		}
 		if needToDisableLogsSync {
 			int.LogsSyncState = "CANCELLING"
```

Comment:
i get confused when see `int` here and there :D  maybe it can be called `integration` ?